cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,357 @@
+
+  -- Take the last two stake delegators and delegate them to "Abstain".
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain1"
+                          wallet1 (defaultDelegatorStakeKeyPair 2)
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain2"
+                          wallet2 (defaultDelegatorStakeKeyPair 3)
+
+  -- Do some other proposal and vote yes with first DRep only
+  -- and assert the new proposal passes now.
+  let newNumberOfDesiredPools2 = fromIntegral (newNumberOfDesiredPools + 1)
+  void $ desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo gov "secondProposal"
+                                       wallet0 Nothing [(1, "yes")] newNumberOfDesiredPools2 newNumberOfDesiredPools2 2
+
+delegateToAlwaysAbstain
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m)
+  => H.ExecConfig
+  -> EpochStateView
+  -> FilePath
+  -> FilePath
+  -> ShelleyBasedEra ConwayEra
+  -> FilePath
+  -> String
+  -> PaymentKeyInfo
+  -> StakingKeyPair
+  -> m ()
+delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe work prefix
+                        payingWallet skeyPair@(StakingKeyPair vKeyFile _sKeyFile) = do
+
+  let era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+
+  baseDir <- H.createDirectoryIfMissing $ work </> prefix
+
+  -- Create vote delegation certificate
+  let voteDelegationCertificatePath = baseDir </> "delegation-certificate.delegcert"
+  void $ H.execCli' execConfig
+    [ "conway", "stake-address", "vote-delegation-certificate"
+    , "--always-abstain"
+    , "--stake-verification-key-file", vKeyFile
+    , "--out-file", voteDelegationCertificatePath
+    ]
+
+  -- Compose transaction to publish delegation certificate
+  repRegTxBody1 <- createCertificatePublicationTxBody execConfig epochStateView sbe baseDir "del-cert-txbody"
+                                                      (File voteDelegationCertificatePath) payingWallet
+
+  -- Sign transaction
+  repRegSignedRegTx1 <- signTx execConfig cEra baseDir "signed-reg-tx"
+                               repRegTxBody1 [ SomeKeyPair (paymentKeyInfoPair payingWallet)
+                                             , SomeKeyPair skeyPair]
+
+  -- Submit transaction
+  submitTx execConfig cEra repRegSignedRegTx1
+
+  -- Wait two epochs
+  (EpochNo epochAfterProp) <- getCurrentEpochNo epochStateView
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterProp + 2))
+
+desiredPoolNumberProposalTest
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m, Foldable t)
+  => H.ExecConfig
+  -> EpochStateView
+  -> FilePath
+  -> FilePath
+  -> ConwayEraOnwards ConwayEra
+  -> FilePath
+  -> FilePath
+  -> PaymentKeyInfo
+  -> Maybe (String, Word32)
+  -> t (Int, String)
+  -> Integer
+  -> Integer
+  -> Integer
+  -> m (String, Word32)
```

Comment:
Can you add haddocks here? Figuring out what is what from the variable names is a bit inconvenient

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/e82f9f3405d9a6e7df8c34fd4eef4ee94d47a11f